modules/geosqlbackend/README.md

```diff
@@ -0,0 +1,598 @@
+
+> **Default** : pdns
+
+
+#### Option: pdns-password
+
+> **Description** : The PowerDNS Database password
+
+> **Default** : pdns
+
+---
+
+**The following options support variable substitution of the following strings:**
+
+| Variable          | Replaced by
+|:------------------|:----------------------------------------------------------------------------------
+| {{REGION}}        | Replaced by the detected region or country code
+| {{DOMAIN-SUFFIX}} | Replaced by domain suffix configured with the "domain-suffix" configuration option
+| {{QDOMAIN}}       | Replaced by the requested domainname (e.g: example.com)
+| {{QTYPE}}         | Replaced by the requested query type (e.g: MX,A, CNAME, TXT, etc...)
+| {{S-IP}}          | Replaced by the IP Address of the original requestor or the ECS Information
+
+
+#### Option: sql-pdns-lookuptype
+
+> **Description** : SQL Statement to retrieve RR types such as A,CNAME,TXT or MX records.
+
+> **Expected Columns** : name, type, content, ttl, prio (order is not important)
+
+> **Default** : select replace(name, '.{{REGION}}.{{DOMAIN-SUFFIX}}','') as name, type , replace(content,'.{{REGION}}.{{DOMAIN-SUFFIX}}','') as content, ttl, prio from records where name='{{QDOMAIN}}.{{REGION}}.{{DOMAIN-SUFFIX}}' and type='{{QTYPE}}' and disabled=0;
+
+
+#### Option: sql-pdns-lookuptype-any
+
+> **Description** : SQL Statement to retrieve the ANY RR type requests
+
+> **Expected Columns** : name, type, content, ttl, prio (order is not important)
+
+> **Default** : select replace(name, '.{{REGION}}.{{DOMAIN-SUFFIX}}','') as name, type, replace(content,'.{{REGION}}.{{DOMAIN-SUFFIX}}','') as content, ttl, prio from records where name='{{QDOMAIN}}.{{REGION}}.{{DOMAIN-SUFFIX}}' and type != 'SOA' and disabled=0;
+
+
+#### Option: sql-geo-lookup-region
+
+> **Description** : SQL Statement to lookup the REGION and Country Code by source IP address
+
+> **Expected Columns** : cc, regionname (order is not important)
+
+> **Default** : select cc,regionname from lookup where MBRCONTAINS(ip_poly, POINTFROMWKB(POINT(INET_ATON('{{S-IP}}'), 0)));
+
+
+
+License
+-------
+
+GeoSQL backend for PowerDNS to support geo-location based DNS responses
+
+Copyright (C) Shin Sterneck 2013-2015 (email: shin at sterneck dot asia)
+
+This program is free software; you can redistribute it and/or
+modify it under the terms of the GNU General Public License
+as published by the Free Software Foundation; either version 2
+of the License, or (at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program; if not, write to the Free Software
+Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
```

Comment:
this should at the very least be gpl2+openssl exception, as all the other powerdns parts